POEApi.Transport/HttpTransport.cs

```diff
@@ -86,23 +86,15 @@ public bool Authenticate(string email, SecureString password, bool useSessionID)
                     TraditionalSessionIdLogin();
                     return true;
                 }
-                catch (Exception ex)
+                catch (WebException ex)
                 {
-                    //Something that isn't a cloud flare exception occurred
-                    if (ex.Message.Contains("503") == false)
-                    {
-                        throw;
-                    }
-
-                    try
+                    if (((HttpWebResponse)ex.Response).Server == "cloudflare" && ((HttpWebResponse)ex.Response).StatusCode == HttpStatusCode.ServiceUnavailable)

```

Comment:
Nit: split this condition across two lines (too long right now).

Reply:
I'm away from my development machine until after the weekend. I'll leave this for now.